### PR TITLE
Fc arm protect

### PR DESCRIPTION
**Pull request by fwc4838**
ARM PROTECTIONS! IT DOESN'T BREAK NOW

### CODE REVIEW

File: src/main/java/frc/robot/commands/extender/ManualMoveExtender.java [hunk 1]
```diff
@@ -13,14 +13,13 @@ public class ManualMoveExtender extends CommandBase {
...
+        addRequirements(extender);
     }
 
     @Override
     public void execute() {
-        double value = doubleSupplier.getAsDouble();
-        // Flipping sign, "down" is positive
-        this.extender.move(-value);
+        double speed = doubleSupplier.getAsDouble();
+        extender.move(speed);
```
Comment (bengold22) [inline src/main/java/frc/robot/commands/extender/ManualMoveExtender.java:22]: You flipped the sign back here, was this intentional?

File: src/main/java/frc/robot/subsystems/Arm.java [hunk 2]
```diff
@@ -67,12 +87,17 @@ public void setPidding(boolean bool) {
   }
 
   public void setVoltage(Double val) {
-    if (val > Constants.ARM_MAX_VOLTS) {
-    neoMotor.setVoltage(Constants.ARM_MAX_VOLTS);
-  } else if (val < -Constants.ARM_MAX_VOLTS) {
-    neoMotor.setVoltage(-Constants.ARM_MAX_VOLTS);}
-    else {
-      neoMotor.setVoltage(val);
+    if (val < 1 && !safeToExtend() && !extender.safeToLowerArm()) {
```
Comment (bengold22) [inline src/main/java/frc/robot/subsystems/Arm.java:101]: Shouldn't this be val < 0, not val < 1? You want to make this check only when it's going down?

Comment (bengold22) [reply src/main/java/frc/robot/subsystems/Arm.java:101]: So as written this logic kicks in whenever the arm is relatively low, and extender is relatively extended, correct?

**Review by bengold22 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/commands/arm/HoldArmPID.java [hunk 3]
```diff
@@ -12,34 +12,22 @@ public class HoldArmPID extends CommandBase {
 
     private Arm arm;
     private Double angle;
```
Comment (armadilloz) [inline src/main/java/frc/robot/commands/arm/HoldArmPID.java:14]: Is this command going to be deleted?

File: src/main/java/frc/robot/commands/arm/ManualMoveArm.java [hunk 4]
```diff
@@ -21,8 +22,8 @@ public void initialize() {
 
     @Override
     public void execute() {
-        // Flipping sign, "down" is positive
-        this.arm.setVoltage(power);
+        arm.setVoltage(power);
```
Comment (armadilloz) [inline src/main/java/frc/robot/commands/arm/ManualMoveArm.java:25]: Is this command used?

File: src/main/java/frc/robot/commands/arm/VoltageMoveArm.java [hunk 5]
```diff
@@ -27,7 +27,7 @@ public void initialize() {
 
     @Override
     public void execute() {
-        //negative value moves arm up
+        //positive angle -> positive power
```
Comment (armadilloz) [inline src/main/java/frc/robot/commands/arm/VoltageMoveArm.java:30]: Is this command used?

File: src/main/java/frc/robot/commands/drive/Move.java [hunk 6]
```diff
@@ -5,39 +5,41 @@
...
         addRequirements(drivetrain);
-    
     }
     
     @Override
     public void end(boolean interrupted) {
         drivetrain.setPower(Constants.DRIVE_BACK_LEFT_D, 0);
         drivetrain.setPower(Constants.DRIVE_BACK_RIGHT_D, 0);
-        //drivetrain.setPower(Constants.DRIVE_FRONT_LEFT_D, 0);
+        drivetrain.setPower(Constants.DRIVE_FRONT_LEFT_D, 0);
```
Comment (armadilloz) [inline src/main/java/frc/robot/commands/drive/Move.java:23]: Shouldn't there be a drivetrain.stop() command?

File: src/main/java/frc/robot/commands/extender/ExtendToPosition.java [hunk 7]
```diff
@@ -31,12 +32,11 @@ public void execute(){
...
-        else {
+        } else {
             speed = Constants.EXTENDER_MINIMUM_SPEED * Math.signum(error);
-            extender.move(speed);
         }
+        
+        extender.move(speed);
     }
 
     @Override
```
Comment (armadilloz) [inline src/main/java/frc/robot/commands/extender/ExtendToPosition.java:42]: Add timeout

**Review by armadilloz — CHANGES_REQUESTED** [review]

**Pull request merged**